Vostok.Metrics.System/Helpers/MetricCollectorHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Vostok.Metrics.Models;
+
+// ReSharper disable PossibleMultipleEnumeration
+
+namespace Vostok.Metrics.System.Helpers
+{
+    internal static class MetricCollectorHelper
```

Comment:
This looks rather complex and a fair bit overengineered. We don't really need support for arbitrary collections here, I reckon it would be much simpler to just handle two dictionaries in HostMetrics separately on top of simple code that existed previously.